### PR TITLE
feat: ensure single component for weights

### PR DESCRIPTION
**Pull request by sjsrey**

### CODE REVIEW

**Comment by sjsrey** [comment]
I'm [pinning to the specific commit in spopt](https://github.com/spatialucr/geosnap/pull/325/files#diff-4d7c51b1efe9043e44439a949dfd92e5827321b34082903477fd04876edb7552R23) for now - the tests pass locally, but clearly this isn't working on the remote ci.

We could cut a release of `spopt` and then remove the pinning?

**Comment by knaaptime** [comment]
the CI is installing from the [yaml file](https://github.com/spatialucr/geosnap/blob/master/.ci/39.yml), so it's not catching this pin, but doing an spopt release would be good

File: requirements.txt [hunk 1]
```diff
@@ -20,7 +20,7 @@ contextily
 descartes
 scikit-plot
 tobler>=0.4
-git+ssh://git@github.com/sjsrey/spopt.git@eef58e64429ff4bc25ed2bf1d715b4ed3cc284ef
+spopt>=0.1.1
```
Comment (knaaptime) [inline requirements.txt:23]: we'll also need to update pin in environment.yml and .ci/ . Tests will still fail for a couple hours until anaconda rebuilds spopt

**Comment by knaaptime** [comment]
seems to be failing due to https://github.com/conda-forge/fontconfig-feedstock/issues/54

should be resolved shortly, i think

**Pull request merged**